provision/kubernetes/provisioner_test.go

```diff
@@ -833,6 +833,75 @@ func (s *S) TestDeploy(c *check.C) {
 	c.Assert(units, check.HasLen, 1)
 }
 
+func (s *S) TestRollback(c *check.C) {
+	a, wait, rollback := s.mock.DefaultReactions(c)
+	defer rollback()
+	deployEvt, err := event.New(&event.Opts{
+		Target:  event.Target{Type: event.TargetTypeApp, Value: a.GetName()},
+		Kind:    permission.PermAppDeploy,
+		Owner:   s.token,
+		Allowed: event.Allowed(permission.PermAppDeploy),
+	})
+	c.Assert(err, check.IsNil)
+
+	customData := map[string]interface{}{
+		"processes": map[string]interface{}{
+			"web": "run mycmd arg1",
+		},
+	}
+	err = image.SaveImageCustomData("tsuru/app-myapp:v1", customData)
+	c.Assert(err, check.IsNil)
+	img, err := s.p.Deploy(a, "tsuru/app-myapp:v1", deployEvt)
+
+	c.Assert(err, check.IsNil)
+	c.Assert(img, check.Equals, "tsuru/app-myapp:v1")
+	deployEvt.Done(err)
+
+	deployEvt, err = event.New(&event.Opts{
```

Comment:
@gfleury Sorry I made a mistake, you need to back the second customData definition, but you can remove the second deployEvt.

in the second customData use something different like `run mycmd arg2` then the last test checking the web command in running unit will match the app:v1